frontend/model/featureCheck.js

```diff
@@ -0,0 +1,15 @@
+const hasFetch = typeof fetch === 'function' && typeof Request === 'function' && typeof Response === 'function'
+const hasReadableStream = typeof ReadableStream === 'function'
+const hasWebSocket = typeof WebSocket === 'function'
+const hasReflect = typeof Reflect === 'object'
+const hasNavigator = typeof Navigator === 'function' && typeof navigator === 'object' && navigator instanceof Navigator
+const hasServiceWorker = hasNavigator && typeof ServiceWorkerContainer === 'function' && navigator.serviceWorker instanceof ServiceWorkerContainer
+// $FlowFixMe[cannot-resolve-name]
+const hasIndexedDB = typeof IDBFactory === 'function' && typeof indexedDB === 'object' && indexedDB instanceof IDBFactory
+const hasSessionStorage = typeof Storage === 'function' && typeof sessionStorage === 'object' && sessionStorage instanceof Storage
+
+console.info('Feature check', { hasFetch, hasReadableStream, hasWebSocket, hasReflect, hasNavigator, hasServiceWorker, hasIndexedDB, hasSessionStorage })
+
+const hasAllRequiredFeatures: boolean = hasFetch && hasReadableStream && hasWebSocket && hasReflect && hasNavigator && hasServiceWorker && hasIndexedDB && hasSessionStorage
```

Comment:
I like this suggestion by o4-mini (high), as it avoids risking missing a variable, looks more elegant, makes logging the console.info more straightforward and DRY, and also is easier to extend in the future with other feature checks:

```js
const requiredFeatures = {
  fetch: typeof fetch === 'function',
  streams: typeof ReadableStream === 'function',
  webSocket: typeof WebSocket === 'function',
  reflect: typeof Reflect === 'object',
  navigator: typeof navigator === 'object' && 'serviceWorker' in navigator,
  indexedDB: typeof indexedDB === 'object',
  sessionStorage: typeof sessionStorage === 'object'
}

const hasAllRequiredFeatures: boolean =
  Object.values(requiredFeatures).every(Boolean)

export default hasAllRequiredFeatures
```

Reply:
Sure, but this is perhaps a bit slower (due to the `Object.values.every`) and mine also checks for consistency. For example, `typeof Storage === 'function' && typeof sessionStorage === 'object' && sessionStorage instanceof Storage` checks for the presence of sessionStorage as well as whether the API is what one expects.

Should I take this mostly as is or add the extra checks?